R/zchunk_L244.building_det.R

```diff
@@ -0,0 +1,1040 @@
+             inc.year.fillout = min(BASE_YEARS),
+             subregional.population.share = 1,
+             subregional.income.share = 1)
+
+    # Internal gains
+    # L244.PriceExp_IntGains: price exponent on floorspace and naming of internal gains trial markets
+    L244.PriceExp_IntGains <- write_to_all_regions(A44.gcam_consumer, LEVEL2_DATA_NAMES[["PriceExp_IntGains"]],
+                                                   GCAM_region_names = GCAM_region_names)
+
+    # L244.Floorspace: base year floorspace
+
+    # Building residential floorspace in the base years
+    # Keep all historical years for now - these are needed in calculating satiation adders later on
+    A44.gcam_consumer_resid <- A44.gcam_consumer %>%
+      filter(grepl("res", A44.gcam_consumer$gcam.consumer))
+
+    L244.Floorspace_resid <- L144.flsp_bm2_R_res_Yh %>%
+      mutate(base.building.size = round(value, energy.DIGITS_FLOORSPACE)) %>%
+      select(-value) %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      mutate(gcam.consumer = A44.gcam_consumer_resid$gcam.consumer,
+             nodeInput = A44.gcam_consumer_resid$nodeInput,
+             building.node.input = A44.gcam_consumer_resid$building.node.input) %>%
+      select(region, gcam.consumer, nodeInput, building.node.input, year, base.building.size)
+
+    # Commercial floorspace
+    A44.gcam_consumer_comm <- A44.gcam_consumer %>%
+      filter(grepl("comm", A44.gcam_consumer$gcam.consumer))
+
+    L244.Floorspace_comm <- L144.flsp_bm2_R_comm_Yh %>%
+      mutate(base.building.size = round(value, energy.DIGITS_FLOORSPACE)) %>%
+      select(-value) %>%
+      left_join_error_no_match(GCAM_region_names, by = "GCAM_region_ID") %>%
+      mutate(gcam.consumer = A44.gcam_consumer_comm$gcam.consumer,
+             nodeInput = A44.gcam_consumer_comm$nodeInput,
+             building.node.input = A44.gcam_consumer_comm$building.node.input) %>%
+      select(region, gcam.consumer, nodeInput, building.node.input, year, base.building.size)
+
+    L244.Floorspace <- bind_rows(L244.Floorspace_resid, L244.Floorspace_comm) %>%
+      filter(year %in% BASE_YEARS)
+
+    # Demand function
+    # L244.DemandFunction_serv and L244.DemandFunction_flsp: demand function types
+    L244.DemandFunction_serv <- write_to_all_regions(A44.demandFn_serv, LEVEL2_DATA_NAMES[["DemandFunction_serv"]],
+                                                     GCAM_region_names = GCAM_region_names)
+    L244.DemandFunction_flsp <- write_to_all_regions(A44.demandFn_flsp, LEVEL2_DATA_NAMES[["DemandFunction_flsp"]],
+                                                     GCAM_region_names = GCAM_region_names)
+
+    # Floorspace demand satiation
+    # L244.Satiation_flsp: Satiation levels assumed for floorspace
+    L244.Satiation_flsp_class <- A44.satiation_flsp %>%
+      gather(sector, value, resid, comm) %>%
+      mutate(satiation.level = value * CONV_THOUS_BIL) %>%
```

Comment:
What unit is this in before and after the conversion?

Reply:
Before: square meters per person. After: million square meters per person. Inconvenient, but it falls out of the model's units for population (thousands) and floorspace (billion m2)